pkg/reconciler/apis/apibinding/apibinding_controller.go

```diff
@@ -69,19 +69,50 @@ func NewController(
 	queue := workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), controllerName)
 
 	c := &controller{
-		queue:                    queue,
-		enqueueAfter:             func(binding *apisv1alpha1.APIBinding, duration time.Duration) { queue.AddAfter(binding, duration) },
-		crdClusterClient:         crdClusterClient,
-		kcpClusterClient:         kcpClusterClient,
-		apiBindingsLister:        apiBindingInformer.Lister(),
-		apiBindingsIndexer:       apiBindingInformer.Informer().GetIndexer(),
-		apiExportsLister:         apiExportInformer.Lister(),
-		apiExportsIndexer:        apiExportInformer.Informer().GetIndexer(),
-		apiResourceSchemaLister:  apiResourceSchemaInformer.Lister(),
+		queue:            queue,
+		enqueueAfter:     func(binding *apisv1alpha1.APIBinding, duration time.Duration) { queue.AddAfter(binding, duration) },
+		crdClusterClient: crdClusterClient,
+		kcpClusterClient: kcpClusterClient,
+
+		apiBindingsLister: apiBindingInformer.Lister(),
+		listAPIBindings: func(clusterName logicalcluster.LogicalCluster) ([]*apisv1alpha1.APIBinding, error) {
+			list, err := apiBindingInformer.Lister().List(labels.Everything())
+			if err != nil {
+				return nil, err
+			}
+
+			var ret []*apisv1alpha1.APIBinding
+
+			for i := range list {
```

Comment:
(no action necessary) I have started adding an index by workspace elsewhere. Is that correct / necessary / sensible? These loops here definitely won't scale long-term.